frontend/src/employee-frontend/components/reports/FuturePreschoolersReport.tsx

```diff
@@ -0,0 +1,202 @@
+// SPDX-FileCopyrightText: 2017-2023 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React, { useEffect, useMemo, useState } from 'react'
+import styled from 'styled-components'
+
+import { Loading, Result } from 'lib-common/api'
+import {
+  FuturePreschoolersReportRow,
+  PreschoolGroupsReportRow
+} from 'lib-common/generated/api-types/reports'
+import Loader from 'lib-components/atoms/Loader'
+import Title from 'lib-components/atoms/Title'
+import ReturnButton from 'lib-components/atoms/buttons/ReturnButton'
+import { Container, ContentArea } from 'lib-components/layout/Container'
+import { defaultMargins } from 'lib-components/white-space'
+
+import {
+  getFuturePreschoolersReport,
+  getPreschoolGroupsReport
+} from '../../api/reports'
+import { useTranslation } from '../../state/i18n'
+
+import ReportDownload from './ReportDownload'
+
+const DownloadWrapper = styled.div`
+  margin-top: calc(-2 * ${defaultMargins.m});
+`
+
+export default React.memo(function FuturePreschoolersReport() {
+  const { i18n } = useTranslation()
+  const [rows, setRows] = useState<Result<FuturePreschoolersReportRow[]>>(
+    Loading.of()
+  )
+  const [municipalRows, setMunicipalRows] = useState<
+    Result<PreschoolGroupsReportRow[]>
+  >(Loading.of())
+  const [voucherRows, setVoucherRows] = useState<
+    Result<PreschoolGroupsReportRow[]>
+  >(Loading.of())
+
+  useEffect(() => {
+    setRows(Loading.of())
+    void getFuturePreschoolersReport().then(setRows)
+  }, [])
+
+  useEffect(() => {
+    setMunicipalRows(Loading.of())
+    void getPreschoolGroupsReport(true).then(setMunicipalRows)
+  }, [])
+
+  useEffect(() => {
+    setVoucherRows(Loading.of())
+    void getPreschoolGroupsReport(false).then(setVoucherRows)
+  }, [])
+
+  const reportRows: FuturePreschoolersReportRow[] = useMemo(
+    () => rows.getOrElse([]),
+    [rows]
+  )
+
+  const municipalGroupRows: PreschoolGroupsReportRow[] = useMemo(
+    () => municipalRows.getOrElse([]),
+    [municipalRows]
+  )
+
+  const privateVoucherGroupRows: PreschoolGroupsReportRow[] = useMemo(
+    () => voucherRows.getOrElse([]),
+    [voucherRows]
+  )
```

Comment:
You should use either `useQueryResult` (newest hot) or at least `useApiState` (a bit older convention) to make the backend requests. You should also use `renderResult()` instead of `.isLoading` etc. to unwrap the `Result` and show a spinner while data is loading.

Unfortunately, most other reports are not good examples in this regard because they're such old code, but you should be able to find plenty of examples in other parts of the code and among more "modern" reports, too.